### DISCUSSION TITLE
Error: Cannot determine the location of the VS Common Tools folder

### DISCUSSION THREAD

**Discussion by tushar95mahawar** [tianocore/edk2#3630] [Q&A]
Hi guys, I am try to build edk setup  but getting Error and warning. If anyone knows please help me to find out the solution... 

C:\edk2>edksetup.bat
ERROR: Cannot determine the location of the VS Common Tools folder.
          PATH      = C:\edk2\BaseTools\Bin\Win32;C;\Nasm;C:\Nasm;C:\edk2\BaseTools\Bin\WIn32;C:\Openssl;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools;C:\Python27;

     WORKSPACE      = C:\edk2\
EDK_TOOLS_PATH      = C:\edk2\BaseTools
BASE_TOOLS_PATH     = C:\edk2\BaseTools
 EDK_TOOLS_BIN      = C:\edk2\BaseTools\Bin\Win32
     CONF_PATH      = C:\edk2\BaseTools\Conf

Using EDK2 in-source Basetools
               PATH = C:\edk2\BaseTools\BinWrappers\WindowsLike;C:\edk2\BaseTools\Bin\Win32;C;\Nasm;C:\Nasm;C:\edk2\BaseTools\Bin\WIn32;C:\Openssl;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools;C:\Python27;
         PYTHONPATH = C:\edk2\BaseTools\Source\Python;C:\edk2\BaseTools\Source\Python

!!! WARNING !!! CLANG_BIN environment variable is not set

**Comment by lersi** [comment]
is the env variable `VS140COMNTOOLS` defined?

the edk script uses that variable to get the location of vs common tools, if it is not defined the search will fail.

the value of `VS140COMNTOOLS` should be `C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\`

**Reply by tushar95mahawar** [reply]
variable value of VS140COMTOOLS is defined in ENV 
that installation path is  C:\devstudio2015\Common7\Tools by default same vlaue i gave in env value

**Reply by lersi** [reply]
sorry for answering late, have you tried to run `edksetup.bat VS2015`?
